client/services/FeedbackGeneratorServiceSpec.js

```diff
@@ -91,5 +98,159 @@ describe('FeedbackGeneratorService', function() {
         "3 seconds) we've set. Can you try to make it run ",
         "more efficiently?"].join(''));
     });
+
+    it(['should return the next hint in sequence for buggy outputs, ',
```

Comment:
There's a fair bit of redundancy, here, and in subsequent tests -- would it be better to make this a global buggyOutputTestDict?